src/main/java/org/update4j/service/DefaultBootstrap.java

```diff
@@ -336,12 +369,14 @@ private static void usage() {
 						+ "\t\tconfiguration file. If it fails to download or command is missing, it will\n"
 						+ "\t\tfall back to local.\n\n"
 						+ "\t--local=[path] - The path of a local configuration to use if the remote failed to download\n"
-						+ "\t\tor was not passed. If both remote and local fail, startup fails.\n\n"
+						+ "\t\tor was not passed. If both remote and local fail, startup fails. You may also use variables e.g. \n"
+						+ "\t\t${stdUserAppData} which refers to the platform dependent application data directory\n\n"
 						+ "\t--syncLocal - Sync the local configuration with the remote if it downloaded successfully.\n"
 						+ "\t\tUseful to still allow launching without Internet connection. Default will not sync unless\n"
 						+ "\t\t--launchFirst was specified.\n\n"
 						+ "\t--cert=[path] - A path to an X.509 certificate file to use to verify signatures. If missing,\n"
-						+ "\t\tno signature verification will be performed.\n\n"
+						+ "\t\tno signature verification will be performed. The path may also point to a included java\n"
+						+ "\t\tresource e.g. res:/app.crt, if the file is embedded in the root directory of the jar\n\n"
```

Comment:
This feature looks like very nice; problem is, update4j will never contain an embedded certificate (other than for testing), and using the default bootstrap does not require any other jars in the boot class/module paths.

And once the developer opts for designing their own bootstrap, you no longer deal with certificates, but public keys (so you can use, say, GPG key-pairs).

Reply:
Well, the user is free to use it and has to package a certificate when using this option. Update4j does not have a standard certificate (and it shouldn't, as the private key would have to be public as it is required for the signatures). The user can simply embed his own Certificate as it is done in the ``gradleExample``:
- Only the ``update4j.jar`` is copied into the installation package (no more files are required to bootstrap the application).
- The ``update4j.jar`` is updated with the users Certificate (``jar uf update4j.jar app.crt`` in task ``bootSignedJar``)
So it will be distributed with the (users) certificate embedded. You can use any X.509 certificate (as supported by the DefaultBootstrap)